### PR TITLE
feat!: separate bindings and allow for custom bindings.

### PR DESCRIPTION
**Pull request by VisualBean**
This PR moves Bindings into their own package (LEGO.AsyncAPI.Bindings)

BREAKING CHANGE: Moves bindings support to their own package.

For Writing, the support is the same, simply add the binding where applicable.
For Reading support has changed, as they arent implicitly added.

Binding support must be added explicitly to the reader through the settings.
A static helper has been made, to make this slightly easier, however.
```csharp
var settings = new AsyncApiReaderSettings();
settings.Bindings.AddRange(BindingsCollection.All);
var reader = new AsyncApiStringReader(settings);
```

### CODE REVIEW

**Comment by dpwdec** [comment]
Nice work man! I guess I will act as a “QA” for this when I integrate it into the AWS bindings work. As ever appreciate you getting so much done on this so quickly! 🙏

File: src/LEGO.AsyncAPI.Readers/Interface/IBindingParser1.cs
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Readers/Interface/IBindingParser1.cs:1]: Is this supposed to be named IBindingParser1? Seems like testing name that was not renamed back

File: src/LEGO.AsyncAPI.Bindings/ChannelBinding{T}.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    using LEGO.AsyncAPI.Readers.ParseNodes;
+
+    public abstract class ChannelBinding<T> : Binding<T>, IChannelBinding
+        where T : IChannelBinding, new()
+    {
+        protected abstract FixedFieldMap<T> FixedFieldMap { get; }
+
+        public override T LoadBinding(PropertyNode node) => BindingDeserializer.LoadBinding("ChannelBinding", node.Value, this.FixedFieldMap);
+    }
+}
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Bindings/ChannelBinding{T}.cs:14]:
```suggestion
}

```

File: src/LEGO.AsyncAPI.Bindings/MessageBinding{T}.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+namespace LEGO.AsyncAPI.Bindings
+{
+    public abstract class MessageBinding<T> : Binding<T>, IMessageBinding
+        where T : IMessageBinding, new()
+    {
+        protected abstract FixedFieldMap<T> FixedFieldMap { get; }
+
+        public override T LoadBinding(PropertyNode node) => BindingDeserializer.LoadBinding("MessageBinding", node.Value, FixedFieldMap);
+    }
+}
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Bindings/MessageBinding{T}.cs:15]:
```suggestion
}

```

File: src/LEGO.AsyncAPI.Bindings/OperationBinding{T}.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+namespace LEGO.AsyncAPI.Bindings
+{
+    public abstract class OperationBinding<T> : Binding<T> , IOperationBinding
+        where T : IOperationBinding, new()
+    {
+        protected abstract FixedFieldMap<T> FixedFieldMap { get; }
+
+        public override T LoadBinding(PropertyNode node) => BindingDeserializer.LoadBinding("OperationBinding", node.Value, FixedFieldMap);
+    }
+}
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Bindings/OperationBinding{T}.cs:16]:
```suggestion
}

```

File: src/LEGO.AsyncAPI.Bindings/Pulsar/PulsarChannelBinding.cs [hunk 4]
```diff
@@ -78,24 +86,8 @@ public void SerializeV2WithoutReference(IAsyncApiWriter writer)
...
-            {
-                this.Reference.SerializeV2(writer);
-                return;
-            }
-
-            this.SerializeV2WithoutReference(writer);
-        }
     }
-}
+}
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Bindings/Pulsar/PulsarChannelBinding.cs:93]:
```suggestion
}

```

File: src/LEGO.AsyncAPI.Bindings/ServerBinding{T}.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+namespace LEGO.AsyncAPI.Bindings
+{
+    public abstract class ServerBinding<T> : Binding<T>, IServerBinding
+        where T : IServerBinding, new()
+    {
+        protected abstract FixedFieldMap<T> FixedFieldMap { get; }
+
+        public override T LoadBinding(PropertyNode node) => BindingDeserializer.LoadBinding("ServerBinding", node.Value, FixedFieldMap);
+    }
+}
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Bindings/ServerBinding{T}.cs:15]:
```suggestion
}

```

File: src/LEGO.AsyncAPI.Bindings/WebSockets/WebSocketsChannelBinding.cs [hunk 6]
```diff
@@ -48,24 +48,8 @@ public void SerializeV2WithoutReference(IAsyncApiWriter writer)
...
-            if (this.Reference != null && !writer.GetSettings().ShouldInlineReference(this.Reference))
-            {
-                this.Reference.SerializeV2(writer);
-                return;
-            }
-
-            this.SerializeV2WithoutReference(writer);
-        }
     }
 }
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Bindings/WebSockets/WebSocketsChannelBinding.cs:55]:
```suggestion
}

```

File: src/LEGO.AsyncAPI.Readers/Interface/IBindingParser1.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+namespace LEGO.AsyncAPI.Readers.Interface
+{
+    using LEGO.AsyncAPI.Models.Interfaces;
+    using LEGO.AsyncAPI.Readers.ParseNodes;
+
+    public interface IBindingParser<out T> : IBinding
+    {
+        T LoadBinding(PropertyNode node);
+    }
+}
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Readers/Interface/IBindingParser1.cs:12]:
```suggestion
}

```

File: src/LEGO.AsyncAPI.Readers/ParseNodes/FixedFieldMap.cs [hunk 8]
```diff
@@ -5,7 +5,7 @@ namespace LEGO.AsyncAPI.Readers.ParseNodes
     using System;
     using System.Collections.Generic;
 
-    internal class FixedFieldMap<T> : Dictionary<string, Action<T, ParseNode>>
+    public class FixedFieldMap<T> : Dictionary<string, Action<T, ParseNode>>
     {
     }
 }
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI.Readers/ParseNodes/FixedFieldMap.cs:11]:
```suggestion
}

```

File: src/LEGO.AsyncAPI/Models/ReferenceType.cs [hunk 9]
```diff
@@ -82,6 +82,10 @@ public enum ReferenceType
...
         /// </summary>
         [Display("headers")] Header,
-        ServerVariable,
+
+        /// <summary>
+        /// The server variable
+        /// </summary>
+        [Display("serverVariable")] ServerVariable,
     }
 }
```
Comment (UlrikSandberg) [inline src/LEGO.AsyncAPI/Models/ReferenceType.cs:91]:
```suggestion
}

```

**Pull request merged**